src/wp-content/themes/twentytwentyfive/functions.php

```diff
@@ -39,19 +39,24 @@ function twentytwentyfive_editor_style() {
 endif;
 add_action( 'after_setup_theme', 'twentytwentyfive_editor_style' );
 
-// Enqueues style.css on the front.
+// Enqueues style.min.css or style.css on the front.
 if ( ! function_exists( 'twentytwentyfive_enqueue_styles' ) ) :
 	/**
-	 * Enqueues style.css on the front.
+	 * Enqueues style.min.css or style.css on the front.
 	 *
 	 * @since Twenty Twenty-Five 1.0
 	 *
 	 * @return void
 	 */
 	function twentytwentyfive_enqueue_styles() {
+		$src = 'style.min.css';
+		if ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG || strpos( wp_get_wp_version(), '-src' ) || ! file_exists( get_parent_theme_file_path( 'style.min.css' ) ) ) {
```

Comment:
In `default-constants.php`, the `SCRIPT_DEBUG` constant is _always_ defined. So we don't need to check for it. This can also use `str_contains()` instead, since the polyfill was introduced in WP 5.9.

```suggestion
		if ( SCRIPT_DEBUG || str_contains( wp_get_wp_version(), '-src' ) || ! file_exists( get_parent_theme_file_path( 'style.min.css' ) ) ) {
```

Reply:
Note that when `SCRIPT_DEBUG` is defined, it will check the version to see if it contains `-src` as well.

Reply:
When considering what core is doing for the emoji loader:

```php
file_get_contents( ABSPATH . WPINC . '/js/wp-emoji-loader' . wp_scripts_get_suffix() . '.js' )
```

I think we can skip the `file_exists()` check as well.

Reply:
If this were using `get_stylesheet_uri()` then we'd need the `file_exists()` check since it could be that the child theme has a `style.css` but not a `style.min.css`. But since we're only looking at the parent theme, then that isn't needed.

Reply:
See https://github.com/WordPress/wordpress-develop/pull/10081#discussion_r2389663315 for a suggestion to tie this together.

Reply:
> Note that when `SCRIPT_DEBUG` is defined, it will check the version to see if it contains `-src` as well.

Okay, `default-constants.php` is handling this, so the check can be removed without an issue.

> If this were using `get_stylesheet_uri()` then we'd need the `file_exists()` check since it could be that the child theme has a `style.css` but not a `style.min.css`. But since we're only looking at the parent theme, then that isn't needed.

I have a doubt about this case. Currently, since we are not shipping minified files in the repo, the contributor needs to navigate to the theme and run `npm run build` to generate the minified files.

Let's say a core contributor is using the TT5 or TT2 theme and sets `SCRIPT_DEBUG` to `false` while debugging some other core scripts TT5 will then try to load the minified file, which doesn’t exist.

There are two solutions I can think of:

1. Keep the file existence check.
2. Create a new Webpack config that generates the minified files when `npm run dev` is run at the root level. (This is required because the Gruntfile only generates the minified files in the build folder.)

Reply:
I believe this is to be expected. For example, if you force `SCRIPT_DEBUG` to be `false` in core without having done a build, you get all kinds of errors related to missing scripts and styles.

Reply:
My bad, after setting `SCRIPT_DEBUG` to `false` and running `npm run dev`, it does generate the minified files for the theme in the `src` directory 😅.